### PR TITLE
add AppVersion column to the history command

### LINKED ISSUE #5044: [FEATURE REQUEST] Allow to show `APP VERSION` column in `helm history` command
**Description**

This is a feature request for allow to detect easily the application version when you want to do a `rollback`.

When I want to do a rollback it's useful for me to know what application version is deployed in the current release, this is something I can see when I execute a `helm list`, for example:
```
helm list
NAME                      	REVISION	UPDATED                 	STATUS  	CHART                      	APP VERSION	NAMESPACE
apisoba-prj-consumer-v21  	12       	Mon Dec 10 17:30:24 2018	DEPLOYED	projection-consumer-0.1.0  	21.0.2     	apisoba-prj-v21
articles-worker           	3       	Tue Dec 11 12:27:20 2018	DEPLOYED	articles-worker-1.1.0      	1.6.0      	articles-worker
noodle                    	14      	Tue Dec 11 15:28:41 2018	DEPLOYED	noodle-1.1                 	1.1613.0   	noodle-v1
storage-classes           	10      	Mon Dec 10 18:26:54 2018	DEPLOYED	storage-classes-0.1.2      	1.0        	storage-classes
```
But the `APP VERSION` is very useful for me when I want to do a rollback, and I'm missing it in the `helm history` command, for example:

**Current Output**
```
helm history noodle --max 5
REVISION	UPDATED                 	STATUS    	CHART      	DESCRIPTION
10      	Tue Dec 11 14:09:26 2018	SUPERSEDED	noodle-1.1 	Upgrade complete
11      	Tue Dec 11 14:35:32 2018	SUPERSEDED	noodle-1.1	Upgrade complete
12      	Tue Dec 11 14:50:04 2018	SUPERSEDED	noodle-1.1	Upgrade complete
13      	Tue Dec 11 14:58:11 2018	SUPERSEDED	noodle-1.1	Upgrade complete
14      	Tue Dec 11 15:28:41 2018	DEPLOYED  	noodle-1.1	Upgrade complete
```
**Expected Output**
```
helm history noodle --max 5
REVISION	UPDATED                 	STATUS    	CHART       APP VERSION	DESCRIPTION
10      	Tue Dec 11 14:09:26 2018	SUPERSEDED	noodle-1.1	1.1609.0    Upgrade complete
11      	Tue Dec 11 14:35:32 2018	SUPERSEDED	noodle-1.1	1.1610.0    Upgrade complete
12      	Tue Dec 11 14:50:04 2018	SUPERSEDED	noodle-1.1	1.1611.0    Upgrade complete
13      	Tue Dec 11 14:58:11 2018	SUPERSEDED	noodle-1.1	1.1612.0    Upgrade complete
14      	Tue Dec 11 15:28:41 2018	DEPLOYED  	noodle-1.1	1.1613.0    Upgrade complete
```
If you don't want to add this column by default it would be very useful to show it via a new flag, for example `--show-app-version`

### PR DESCRIPTION
**Pull request by adshmh**
**What this PR does / why we need it**:
Adds an `AppVersion` column to the output of the history command. Closes #5044 

**Special notes for your reviewer**:
While adding unit tests for this change, I noticed that a unit test for the history command does not seem to cover the max option that it targets. Added a commit to fix the unit test to cover the max option and add the required support to the fake client.

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by bacongobbler — APPROVED** [review]
needs a review from another core maintainer but LGTM

**Review by mattstep — APPROVED** [review]

**Review by mattfarina — APPROVED** [review]

**Comment by mattfarina** [comment]
The merge I performed was just to update a generated timestamp. A simple conflict.

**Pull request merged**
